doom-modeline-segments.el

```diff
@@ -1430,6 +1432,32 @@ mouse-1: Display Line and Column Mode Menu"
                                 'mode-line-inactive))))
 
 
+;;
+;; objed-state
+;;
+
+(defvar doom-modeline--objed-active nil)
+
+(defun doom-modeline-update-objed (_ &optional reset)
+  "Update `objed' status, inactive when RESET is true."
+  (setq doom-modeline--objed-active (not reset)))
+
+(setq objed-modeline-setup-func #'doom-modeline-update-objed)
+
+(doom-modeline-def-segment objed-state ()
+  "The current objed state."
+  (when (and doom-modeline--objed-active
```

Comment:
Is there any check here? Like `objed-mode` or `objed-local-mode`? I am not sure if `doom-modeline--objed-active` is safe enough.

Reply:
The module provides a modeline setup hook `objed-modeline-setup-func`, which is called to set the modeline on entry to the modal state, and called again with the reset flag when exiting.  Normally this runs a routine to modify the modeline directly, but we update a variable for the segment instead.

Ideally `objed` would provide a state indicator variable that could be used for this purpose, but it doesn't.  Other internal variables can't be relied upon as they're often left with the previous values even when leaving the modal state.  I don't mind submitting some improvements upstream, but the segment works reliably as-is in my testing for the last week.